### ISSUE TITLE
[dandelion] specify node address for dandelion stem phase

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin#1250]
Right now its *really* difficult to test/demonstrate the aggregation behavior via Dandelion.
You can send txs from a couple of nodes but the chances of them actually crossing paths during the stem phase is low.

* we only send txs during stem phase to a single outbound node
* we randomly choose that stem node every 10 mins
* we need to wait for a confirmation before sending a subsequent tx (spend all outputs by default)

It would be useful if we could either - 
1. configure a single outbound node in `grin.toml` to always forward stems txs to, or
1. pass an optional (first hop) node address when sending a tx from the grin command line

This would allow us to get a couple of txs into the same node within a short period of time and have them aggregate on their way through the stem phase with reasonably high success rate.

**Comment by beaver-tooth** [comment]
Hey I am interested in giving this a whack this weekend.

**Comment by sesam** [comment]
To get warmed up, you could start with adding a command line option and
print it out like "Dandelion stem added: (partially masked IP)" so it will
show in the logs, but not fully give away where the stem is. I.e. logs
should not give away too much details if grin wants to claim the anonymous
niche.
And on Monday there's a dev meeting on gitter chat, please come join there!

**Comment by beaver-tooth** [comment]
Gave this a start. @antiochp are you partial to option 1 or 2?
> 
>    1. configure a single outbound node in grin.toml to always forward stems txs to, or
>    2. pass an optional (first hop) node address when sending a tx from the grin command line
> 

Or would you like to see a PR with both?

**Comment by beaver-tooth** [comment]
Thanks @sesam , I will go down this route. I will also try to tune in monday!

**Comment by beaver-tooth** [comment]
Just wanted to update -- I spent last couple days getting grin up and running and also figuring out how to get a proper local net up and working to play around.

With this up and running, I am able to give this a serious start.

**Comment by beaver-tooth** [comment]
Giving this up, swamped with outside work stuff right now.

**Comment by TTD1993** [comment]
grin.toml

**Comment by svechinsky** [comment]
Is this still necessary?
If so I would like to give it a try

**Comment by sesam** [comment]
yes I think so. There might be other things more urgent though until mainnet. Once you got grin building and synched up, come into the dev chat for a better update.
